driver.go

```diff
@@ -74,5 +74,14 @@ func NewWithDSN(dsn string) (*sql.DB, Sqlmock, error) {
 	pool.conns[dsn] = smock
 	pool.Unlock()
 
-	return smock.open()
+	return smock.open(options)
+}
+
+// WithValueConverter allows to create a sqlmock connection
+// with a custom ValueConverter to support drivers with special data types.
+func WithValueConverter(converter driver.ValueConverter) func(*sqlmock) error {
```

Comment:
can we create the `options.go` file and rename this to `ValueConverterOption`, then move it to `options.go` file ? I think it will be less confusing. Also the **godoc** needs an update for `New` and `NewWithDSN` methods, since now it will have options.

I will later probably extract the DSN to option too. and deprecate the NewWithDSN method.

Reply:
Done.